src/app/config/config.c

```diff
@@ -457,6 +457,7 @@ static config_var_t option_vars_[] = {
   VAR("HiddenServiceMaxStreams",LINELIST_S, RendConfigLines, NULL),
   VAR("HiddenServiceMaxStreamsCloseCircuit",LINELIST_S, RendConfigLines, NULL),
   VAR("HiddenServiceNumIntroductionPoints", LINELIST_S, RendConfigLines, NULL),
+  VAR("HiddenServiceExportCircuitID", LINELIST_S,  RendConfigLines, NULL),
```

Comment:
This option seems to be about controlling if we export the circuit ID in the `HAProxy` format... What about we make it a bit more specific to HAProxy?

Reply:
I think the feature should be independent of the HAProxy implementation. For now we *only* support HAProxy here (and we might only do that for a long time?), but maybe there are other systems like that we could do (for Freenode's v3 IRCd for example?). Maybe it could make sense to have a flag here so the config line in `torrc` becomes something like:

`HiddenServiceExportCircuitID HAProxy` ?

Reply:
@ahf here is a comment by roger that de-suggests that: https://trac.torproject.org/projects/tor/ticket/4700#comment:17

Not that it matters, but some extra feedback.